src/main/java/com/openshift/jenkins/plugins/pipeline/OpenShiftImageStreams.java

```diff
@@ -149,49 +153,66 @@ public SCMRevisionState calcRevisionsFromBuild(Run<?, ?> build,
     	String commitId = lastCommitId;
 			
 		ImageStreamRevisionState currIMSState = null;
+		if (commitId == null) {
+			commitId = this.getCommitId(listener, build.getEnvironment(listener));
+		}
+
 		if (commitId != null) {
 			currIMSState = new ImageStreamRevisionState(commitId);
 			listener.getLogger().println(String.format(MessageConstants.SCM_LAST_REV, currIMSState.toString()));
 		} else {
-	    	listener.getLogger().println(MessageConstants.SCM_NO_REV);
+			listener.getLogger().println(MessageConstants.SCM_NO_REV);
 		}
-		
+
 		
 		return currIMSState;
 	}
 
-	@Override
-	protected PollingResult compareRemoteRevisionWith(
-			AbstractProject<?, ?> project, Launcher launcher,
-			FilePath workspace, TaskListener listener, SCMRevisionState baseline)
-			throws IOException, InterruptedException {
-    	listener.getLogger().println(String.format(MessageConstants.SCM_COMP, DISPLAY_NAME, getImageStreamName(project.getEnvironment(null, listener)), tag, namespace));
-    	String commitId = this.getCommitId(listener, project.getEnvironment(null,listener));
-		
+	protected PollingResult compareRemoteRevisionInternal(EnvVars env, Launcher launcher, TaskListener listener, SCMRevisionState baseline) {
```

Comment:
So I tested different scenarios. When used within a pipeline, the PROJECT_NAME variable is not present. However, I created a regular build job with the ImageStreams SCM and the PROJECT_NAME was present in that case.